sphinx/registry.py

```diff
@@ -281,11 +281,13 @@ def create_source_parser(self, app, filename):
             parser.set_application(app)
         return parser
 
-    def add_source_input(self, filetype, input_class):
-        # type: (unicode, Type[Input]) -> None
-        if filetype in self.source_inputs:
-            raise ExtensionError(__('source_input for %r is already registered') % filetype)
-        self.source_inputs[filetype] = input_class
+    def add_source_input(self, input_class):
```

Comment:
Doesn't this need a deprecation period and release note, given that we're changing behaviour on a public interface?

Reply:
We should probably rename `app.registry` to `app._registry` to emphasise this.

Personally, I'm fine with just dropping it. I'm just making sure we're not doing anything wrong, heh

Reply:
Python does not have strict access control, so renaming is not effect. In fact, some developers had accessed underscored members and methods once.
So I feel it is meaningless so much. And I personally prefer not to use underscored methods.

(I know it declares the methods and members are private obviously.)